File: solution/4375/main.py
def main():
    input = open(0, "rb").readline
    while True:
        try:
            n = int(input())
            length = 1
            remainder = 1 % n
            while remainder != 0:
                remainder = (remainder * 10 + 1) % n
                length += 1
            print(length)
        except ValueError:
            break


if __name__ == "__main__":
    main()

File: solution/4375/Main.java
import java.io.*;

public class Main {
    static BufferedReader stdin = new BufferedReader(new InputStreamReader(System.in), 1048576);
    static StringBuilder stdout = new StringBuilder();
    public static void main(String[] args) throws Exception {
        String line;
        while (true) {
            line = stdin.readLine();
            if (line == null) break;
            int n = Integer.parseInt(line);
            int length = 1;
            int remainder = 1 % n;
            while (remainder != 0) {
                remainder = (remainder * 10 + 1) % n;
                length++;
            }
            stdout.append(length).append('\n');
        }
        System.out.println(stdout);
    }
}
